### PR TITLE
DB connection rework and integration with sda-pipeline

### PR DESCRIPTION
**Pull request by blankdots**
- reuse DB connection
- small list of unit tests - more tests will be pushed in a follow-up PR.

### CODE REVIEW

File: sda_orchestrator/utils/db_ops.py [hunk 1]
```diff
@@ -1,9 +1,39 @@
...
+from typing import Dict
 import psycopg2
 from .logger import LOG
 from time import sleep
 import os
 from pathlib import Path
+from psycopg2.extras import DictCursor
+
+
+class DB:
```
Comment (pontus) [inline sda_orchestrator/utils/db_ops.py:11]: I suspect this doesn't do what I think is intended (e.g. use two different connections).

Comment (blankdots) [reply sda_orchestrator/utils/db_ops.py]: now it should be revamped, and tested. however noticed that if we use mapper this is not really needed.

Comment (blankdots) [reply sda_orchestrator/utils/db_ops.py]: but tested until i hit https://github.com/neicnordic/sda-pipeline/issues/138

File: sda_orchestrator/utils/db_ops.py [hunk 2]
```diff
@@ -1,12 +1,102 @@
...
+
+class DBConnection:
+    """DBConnection class for using singleton db connnection."""
+
+    connection = None
+
+    @classmethod
+    def get_connection(
+        cls: psycopg2.connect, new: bool = False, user: Union[str, None] = None, passwd: Union[str, None] = None
+    ) -> psycopg2.connect:
```
Comment (pontus) [inline sda_orchestrator/utils/db_ops.py:56]: I don't understand why the type here isn't incorrect - psycopg2.connect will be a function and what I'd expect to get passed would be the class (with type `<class 'type'>`).

Comment (blankdots) [reply sda_orchestrator/utils/db_ops.py:56]: I can't say i do either ...  but
https://www.psycopg.org/docs/module.html#psycopg2.connect 
> Create a new database session and return a new connection object.

Comment (blankdots) [reply sda_orchestrator/utils/db_ops.py:56]: or any suggestions ?

**Review by pontus — APPROVED** [review]

**Pull request merged**